File: Java/BOJ/4659 비밀번호 발음하기.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] vowel = {"a", "e", "i", "o", "u"};
        while (true) {
            String password = br.readLine();
            boolean check = true;
            int checknum = 0;

            if (password.equals("end")) {
                break;
            }
            
            int flag = 0;
            for (String i : vowel) {
                if (password.contains(i)) {
                    flag = 1;
                	break;
                }
            }
            if(flag == 0) {
                check = false;
            }
            
            int vt = 0, ct = 0;
            for (int i = 0; i < password.length(); i++) {
                char ch = password.charAt(i);
                if ("aeiou".indexOf(ch) != -1) {
                    vt++;
                    ct = 0;
                } else {
                    vt = 0;
                    ct++;
                }

                if (vt == 3 || ct == 3) {
                    check = false;
                    break;
                }
            }

            for (int i = 0; i < password.length() - 1; i++) {
                if (password.charAt(i) == password.charAt(i + 1) && password.charAt(i) != 'e' && password.charAt(i) != 'o') {
                    check = false;
                    break;
                }
            }

            if (check) {
                System.out.println("<" + password + "> is acceptable.");
            } else {
                System.out.println("<" + password + "> is not acceptable.");
            }
        }   
    }
}

File: Python/BOJ/4659 비밀번호 발음하기.py
import sys

vowel = ['a','e','i','o','u']

while True:
    password = sys.stdin.readline().strip()
    check = True
    checknum = 0
    if password == 'end':
        break

    for i in vowel:
        if i in password:
            break
    else:
        check = False
    
    vt, ct = 0,0
    for i in password:
        if i in vowel :
            vt += 1
            ct = 0
        else :
            vt = 0
            ct += 1

        if vt == 3 or ct == 3:
            check = False
            break

    for i in range(len(password)-1):
        if password[i] == password[i+1] and password[i] != 'e' and password[i] != 'o':
            check = False
            break
    
    if check:
        print(f'<{password}> is acceptable.')
    else:
        print(f'<{password}> is not acceptable.')